docs/guides/gutenberg.md

```diff
@@ -0,0 +1,116 @@
+---
+title: "Gutenberg"
+menu:
+  main:
+    parent: "guides"
+---
+## Using Gutenberg with Timber
+
+Timber works with Gutenberg out of the box. If you use `{{ post.content }}`, Timber will render all the Gutenberg blocks.
+
+## ACF Blocks
+
+### What are ACF Blocks?
+
+ACF Blocks are an alternative way to create content blocks without advanced JavaScript knowledge. If you want to learn more about them, read the article on [advancedcustomfields.com](https://www.advancedcustomfields.com/blog/acf-5-8-introducing-acf-blocks-for-gutenberg/).
+
+### How to use ACF Blocks with Timber
+
+Before you can start using ACF Blocks, you must install the Advanced Custom Fields 5.8.0-beta version or later.
+
+To create a content block, you first have to register it in **functions.php** or in a separate plugin:
+
+```php
+add_action( 'acf/init', 'my_acf_init' );
+function my_acf_init() {
+    // Check function exists.
+    if( function_exists('acf_register_block') ) {
+        
+        // Register a new block.
+        acf_register_block(array(
+            'name'				=> 'example_block',
+            'title'				=> __( 'Example Block', 'your-text-domain' ),
+            'description'		=> __( 'A custom example block.', 'your-text-domain' ),
+            'render_callback'	=> 'my_acf_block_render_callback',
+            'category'			=> 'formatting',
+            'icon'				=> 'admin-comments',
+            'keywords'		    => array( 'example' ),
+        ) );
+    }
+}
+```
+
+Next, you you have to create your `render_callback()` function:
+
+```php
+function my_acf_block_render_callback( $block ) {
+    $context = array();
```

Comment:
This implementation creates a high amount of confusion for developers as the cannot use `theme` or `site` variables afterwards. Maybe use `$context = Timber::get_context()` here? This should not negatively impact performance right @gchtr ?

Reply:
cc @palmiak

Reply:
@pascalknecht very good point. To be honest I was thinking about this but most times when I'm building blocks (Gutenberg or Flexible Content) I'm trying to pass only what's necessary, but your right in terms of user confusion. 

I also think we should add this. Less problems and less issues.

Reply:
In Timber 1.x, you can call `Timber::get_context()` again, without losing performance 👍. So I’d add this as well.

For Timber 2.x, we’d have to use [`Timber::context_global()`](https://github.com/timber/timber/blob/2.x/docs/guides/context.md#context-cache). You could add a second pull request with this change for 2.x once this is merged in. Otherwise, we might forget it again.

Reply:
Pull request submitted #1872!